### PR TITLE
[Feat] Add support for Amazon Games

### PR DESCRIPTION
**Pull request by Gustash**
### Amazon Games Support

This PR adds support for Amazon Games.

It uses [Nile](https://github.com/imLinguin/nile) to interface with your Amazon Games library.

2023-06-30-215042_hyprshot

I have tested on Linux, macOS and Windows. All the basic features have been seen as working, as noted in the TODO list below.

EDIT:

Also tested the .flatpak artifact on the Steam Deck, all features are working there too.

TODO:

- [x] Display list of games
- [x] Filter by Amazon Games only
- [x] Game Info page
- [x] Installing game
- [x] Running game
- [x] Download progress 
- [x] Authentication
- [x] Game Updates
- [x] Importing (?)

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [x] Created / Updated documentation (If necessary)

### CODE REVIEW

**Comment by Gustash** [comment]
Authentication is currently blocked until https://github.com/imLinguin/nile/issues/27 is resolved

**Comment by derrod** [comment]
Currently this appears to not detect if a user has already logged in manually via the CLI. Otherwise seems to work fine for me (on Windows).

**Comment by imLinguin** [comment]
@derrod this is due to sandboxed config of Nile. Same we do with legendary now

**Comment by derrod** [comment]
> @derrod this is due to sandboxed config of Nile. Same we do with legendary now

The config file is migrated still, so Nile was logged in when running from Heroic. But it still showed as logged out and clicking the login button just did nothing. Also noteworthy that even deleting the Nile config didn't work as it would then copy the global one again, thus getting into a bad state again.

**Comment by imLinguin** [comment]
Ah yes. I encountered same issue. Heroic doesn't detect that Nile is logged in already

**Comment by Gustash** [comment]
> Currently this appears to not detect if a user has already logged in manually via the CLI. Otherwise seems to work fine for me (on Windows).

This has been resolved now

**Comment by flavioislima** [comment]
That is pretty cool. Thanks for that.
One comment, not related though, is that we need to figure it out a better way of having those store filters since now it is not that practical. We should change it to be a select with the options instead.

I will test as soon as possible on all platforms.

**Comment by Gustash** [comment]
> That is pretty cool. Thanks for that.

I'm happy to contribute ☺️

> One comment, not related though, is that we need to figure it out a better way of having those store filters since now it is not that practical. We should change it to be a select with the options instead.

Is this something you want me to do on this PR, or would it be more relevant in a separate PR after this?

File: src/common/types.ts [hunk 1]
```diff
@@ -96,7 +98,7 @@ export interface ExtraInfo {
 export type GameConfigVersion = 'auto' | 'v0' | 'v0.1'
 
 export interface GameInfo {
-  runner: 'legendary' | 'gog' | 'sideload'
+  runner: 'legendary' | 'gog' | 'sideload' | 'nile'
```
Comment (flavioislima) [inline src/common/types.ts:101]: We should use `Runner` here at some point.

**Review by flavioislima — COMMENTED** [review]
Testing here. 
One thing that can be improved: 
if there are only windows games on Prime, then we could hide the OS filter. Right now it disables macOS but keeps the browser button, Browser should actually only show when OTHER is selected.
Clicking on it while the amazon is selected does nothing.

**Comment by flavioislima** [comment]
@Gustash 
Let's do the other stuff I said on another PR.
For now just update the nile binaries and fix the conflicts and then it should be ready to merge 👍🏽

**Comment by Gustash** [comment]
@flavioislima I've fixed the merge conflicts. I'm just waiting on Linguin fixing subprocess reaping in nile for Windows and macOS, as it currently doesn't work for killing games from Heroic.

**Comment by Gustash** [comment]
Force stopping games is now working properly for Linux and macOS. On Windows it still doesn't work, but I found that gogdl also doesn't force stop the game on Windows. Since Nile shares pretty much a 1-to-1 mechanism for subprocesses as gogdl, this is not an isolated issue and has probably existed on Windows for a while.

I say we get this PR merged and deal with gogdl and nile not stopping games on Windows in a separate PR.

File: src/backend/storeManagers/nile/games.ts [hunk 2]
```diff
@@ -0,0 +1,591 @@
...
+      ...wineEnvVars
+    }
+
+    const { bin: wineExec, type: wineType } = gameSettings.wineVersion
+
+    // Fix for people with old config
+    const wineBin =
+      wineExec.startsWith("'") && wineExec.endsWith("'")
+        ? wineExec.replaceAll("'", '')
+        : wineExec
```
Comment (imLinguin) [inline src/backend/storeManagers/nile/games.ts:382]: I think this part is not needed, as there is no old config for Amazon games

**Review by imLinguin — APPROVED** [review]
Left small comment, but overall it looks good to me.

Amazing work ⚔️

File: src/backend/storeManagers/nile/games.ts (hunk 2 above)
Comment (Gustash) [reply src/backend/storeManagers/nile/games.ts:382]: Are we sure that global configs can't trigger this case?

Comment (imLinguin) [reply src/backend/storeManagers/nile/games.ts:382]: hmm, maybe you're right

**Pull request merged**